src/plugins/intel_cpu/src/emitters/snippets/x64/jit_loop_emitters.cpp

```diff
@@ -13,93 +14,58 @@ using namespace dnnl::impl::cpu::x64;
-    : jit_loop_begin_emitter(h, isa, expr), loop_end_label(nullptr) {
-    OV_CPU_JIT_EMITTER_ASSERT(ov::is_type<snippets::op::LoopBeginDynamic>(expr->get_node()), "expects LoopBeginDynamic expression");
-    const auto loop_end = get_loop_end(expr);
-    wa_increment = loop_end->get_increment();
-    loop_id = loop_end->get_id();
+size_t jit_loop_begin_emitter::aux_gprs_count() const {
+    // for runtime arguments
+    return need_runtime_args ? 1 : 0;
 }
 
-void jit_loop_begin_dynamic_emitter::validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
-    // Note: the only expected input is the reg_runtime_params_idx
+void jit_loop_begin_emitter::validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
     OV_CPU_JIT_EMITTER_ASSERT(in.empty(), "Invalid inputs size: expected 0 got " + std::to_string(in.size()));
     // Note: the only expected output is work amount register (communicated to jit_loop_end_emitter)
     OV_CPU_JIT_EMITTER_ASSERT(out.size() == 1, "Invalid outputs size: expected 1 got " + std::to_string(out.size()));
-    OV_CPU_JIT_EMITTER_ASSERT(loop_end_label != nullptr && loop_begin_label != nullptr, "has not inited labels!");
+    OV_CPU_JIT_EMITTER_ASSERT(loop_begin_label != nullptr && loop_end_label != nullptr, "has not inited labels!");
+    OV_CPU_JIT_EMITTER_ASSERT(implication(need_runtime_args, !evaluate_once), "with dynamic work_amount cannot evaluate once!");
 }
 
-void jit_loop_begin_dynamic_emitter::emit_code(const std::vector<size_t> &in, const std::vector<size_t> &out,
-                                               const std::vector<size_t> &pool_vec_idxs, const std::vector<size_t> &pool_gpr_idxs) const {
+void jit_loop_begin_emitter::emit_code(const std::vector<size_t> &in, const std::vector<size_t> &out,
+                                       const std::vector<size_t> &pool_vec_idxs, const std::vector<size_t> &pool_gpr_idxs) const {
     validate_arguments(in, out);
-    jit_emitter::emit_code(in, out);
-}
-
-void jit_loop_begin_dynamic_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
-    Xbyak::Reg64 reg_runtime_params = abi_param1;  // defined by jit_kernel_emitter
-    Xbyak::Reg64 reg_work_amount = Xbyak::Reg64(static_cast<int>(out.back()));
-    Xbyak::Reg64 reg_loop_args_ptr = Xbyak::Reg64(static_cast<int>(aux_gpr_idxs[0]));
-    const auto id_offset = loop_id * sizeof(jit_snippets_call_args::loop_args_t);
-    h->mov(reg_loop_args_ptr, h->ptr[reg_runtime_params + GET_OFF(loop_args)]);
-    h->mov(reg_work_amount, h->ptr[reg_loop_args_ptr + id_offset + GET_OFF_LOOP_ARGS(m_work_amount)]);
+    jit_emitter::emit_code(in, out, pool_vec_idxs, pool_gpr_idxs);
+}
+
+void jit_loop_begin_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
+    Reg64 reg_work_amount = Reg64(static_cast<int>(out.back()));
```

Comment:
I think we cannot since `evaluate_once = true` is possible only for static loop (it's static loop optimization to not compile many instructions).
As you can see at L55:  we compile some instructions for Loop with dynamic `work_amount`
So we cannot return early here

Reply:
In general I'm not sure that we should implement `evaluate_once = true` for loop with static `work_amount` and dynamic pointer data shifts: for that we should make additional steps in runtime configurator (add check if `work_amount == 1`, let's sum pointer increments and finalization offsets. But do we really need it? It's open question for me 😃 

If you have some ideas, welcome!

Reply:
Discussed offline early return: agree with suggestion
Implemented in https://github.com/openvinotoolkit/openvino/pull/24525/commits/99a3f713d3243a9c42c7d96b2eebb8efc5d30222